src/shared/components/copyDownloadControls/CopyDownloadControls.tsx

```diff
@@ -102,6 +103,16 @@ export class CopyDownloadControls extends React.Component<
         );
     }
 
+    /**
+     * Wrapper around downloadData() to return as a Promise<string> for ICopyDownloadButtonsProps
+     * see TECH_DOWNLOADDATA
+     */
+    private downloadDataAsStringAsync(): Promise<string> | undefined {
+        if (this.props.downloadData) {
+            return this.props.downloadData().then(data => data.text);
+        }
+    }
+
```

Comment:
The click handler for CustomButton needs the actual data, so downloadData() is passed to the component. There are two ways that buttons get data though, synchronous (SimpleCopyDownloadControls) and asynchronous (CopyDownloadControls). So we set pass downloadData as a Promise<string> then we need have a wrapper to convert the synchronous case to a Promise<string>.  That is this method.

Comments related to this are tagged as "TECH_DOWNLOADDATA".

OPTIONAL: technically, CustomButton could take a Promise<ICopyDownloadData>, but then we would have to move that interface out of CopyDownloadControls.tsx and I wanted to minimize the changes.